### PR TITLE
Remove mapFileBs to eliminate bytestring dependency

### LINKED ISSUE #166: Untie Win32 from bytestring
I am working on refactoring `bytestring` test suite. It is currently organized as two packages. There is a `bytestring` package, which reside in a top-level folder, and a `bytestring-tests` package in `tests` sub-folder with the following configuration:

```cabal
test-suite {foo,bar,baz}
  hs-source-dirs:   . ..
  main-is:          {Foo,Bar,Baz}.hs
  other-modules:    Data.ByteString
                    Data.ByteString.Char8
                    Data.ByteString.Internal
                    Data.ByteString.Lazy
                    Data.ByteString.Lazy.Char8
                    Data.ByteString.Lazy.Internal
                    Data.ByteString.Short
                    Data.ByteString.Short.Internal
                    ...many more...
  build-depends:    base, ghc-prim, deepseq, random,
                    tasty, tasty-quickcheck, QuickCheck
```

So tests do not depend on `bytestring` package explicitly, but rather just happen to share source files. This is unfortunate for many reasons. For example, each test component recompiles all modules of `bytestring` anew. Tooling such as HLS cannot figure out which source file belongs to what. But most importantly this way we do not actually test `bytestring` package, because there are no guarantees that the configuration in `bytestring.cabal` (GHC flags, C sources, etc.) is the same.

An obvious solution is to abandon the hack with `hs-source-dirs: ..` and add `bytestring` as a proper `build-depends` of `bytestring-tests` package. This still does not work in a satisfactory manner, because now any change to `bytestring` causes rebuilding of `random`, `directory`, `QuickCheck`, `process`, `optparse-applicative`, `tasty`, `tasty-quickcheck` (all of them depend on `bytestring` transitively) and, as you can imagine, it takes obnoxiously long for any sort of interactive development.

Ideally I would like to eliminate `bytestring` as a transitive dependency of the test suite. Not only it would enable us to run tests quicker, but also the very reason for having a separate package `bytestring-tests` would disappear.

This idea proved to be fruitful for Linux and MacOS builds: I patched `optparse-applicative` and `tasty` ecosystem so that they no longer depend on `bytestring`, when compiled with right flags. Unfortunately, there is a nasty complication on Windows. The reason is that several packages (e. g., `ansi-terminal`) depend on `Win32`, which depends on `bytestring`, which makes it a circular dependency.

----

I was initially writing this to ask about a possibility to make `bytestring` an optional dependency of `Win32`, guarding relevant calls by a Cabal flag. But actually I found out that there is only a single spot of `Win32`, which involves `bytestring`:

https://github.com/haskell/win32/blob/6775f8ccfd118f76f69e8b447e59f8f81e31a514/System/Win32/FileMapping.hsc#L56-L60

Moreover, this function is unused: neither [Hackage](https://hackage-search.serokell.io/?q=mapFileBs) nor [GHC](https://gitlab.haskell.org/search?group_id=&project_id=&repository_ref=&scope=blobs&search=mapFileBs&snippets=false) refer to it. The code itself is prone to be broken in future, because it relies on an internal API (there are actually ideas to back `ByteString` by `ByteArray#` instead of `ForeignPtr`).

How do you feel about removing `mapFileBs` (and `bytestring` dependency) in the future release of `Win32`?

Not only this would resolve my problem and facilitate smooth testing of `bytestring`, but it would genuinely untie two boot packages from one another.

### PR DESCRIPTION
**Pull request by Bodigrim**
Closes #166.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have not added a new Haskell dependency.
- [x] I have included a changelog entry.
- [x] I have not modified the version of the package in `Win32.cabal`.

I set up GitHub Actions in a separate branch to test my changes: https://github.com/Bodigrim/win32/actions/runs/459405515

### CODE REVIEW

File: System/Win32/FileMapping.hsc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{-# LANGUAGE Trustworthy #-}
```
Comment (Mistuke) [inline System/Win32/FileMapping.hsc:1]: This change is causing older GHCs to fail
```
System/Win32/FileMapping.hsc:25:1:
    Foreign: Can't be safely imported! The module itself isn't safe.
```

Comment (Bodigrim) [reply System/Win32/FileMapping.hsc:1]: Fixed.

**Comment by Mistuke** [comment]
Awesome thanks!

**Comment by Mistuke** [comment]
Do you need an early release or does the end of month releases schedule work for you?

**Comment by Bodigrim** [comment]
This is not immediately pressing, but I'd appreciate an earlier release, if it is not too much trouble. This is hopefully the last piece of my puzzle with `bytestring`.

**Comment by Mistuke** [comment]
Sure, no problem, I'll make the release today

**Comment by Bodigrim** [comment]
@Mistuke just a gentle reminder, in case you have time to spare.

**Comment by Mistuke** [comment]
Oh yeah sorry work kept me busy. Let me do it now

**Comment by Bodigrim** [comment]
Perfect, thanks!

**Pull request merged**